### PR TITLE
Node E2E: add termination message test

### PR DESCRIPTION
**Pull request by Random-Liu**
Based on #23658.

This PR:
1) Cleans up the `ConformanceContainer` a bit
2) Add termination message test

This test proves #23639, without #23658, the test could not pass.

@liangchenye @kubernetes/sig-node

### CODE REVIEW

**Comment by Random-Liu** [comment]
@k8s-bot test this issue #21367

**Comment by pwittrock** [comment]
@Random-Liu Would you check this test failure `test/e2e_node/container.go:51: undefined: nodeName`

**Comment by Random-Liu** [comment]
@pwittrock Just wanna to ask you about that. @chenyeliang met this before. The `nodeName` is just in the same directory.

**Comment by pwittrock** [comment]
Symbols aren't exported from _test.go files :(  If you move it to a non-test file I suspect it will work.

**Comment by Random-Liu** [comment]
Thanks! Good to know that! :)

**Comment by Random-Liu** [comment]
@pwittrock Fixed, :)

File: test/e2e_node/conformance_test.go [hunk 1]
```diff
@@ -99,66 +97,85 @@ var _ = Describe("Container Conformance Test", func() {
...
-				err := terminateCase.Delete()
+			It("should report termination message if TerminationMessagePath is set [Conformance]", func() {
+				terminationMessage := "DONE"
+				terminationMessagePath := "/dev/termination-log"
+				c := NewConformanceContainer(cl, *nodeName, api.Container{
+					Image:   "gcr.io/google_containers/busybox:1.24",
+					Name:    "termination-message-container",
+					Command: []string{"/bin/sh", "-c"},
+					Args:    []string{"sleep 60 && /bin/echo -n " + terminationMessage + " > " + terminationMessagePath},
+					TerminationMessagePath: terminationMessagePath,
```
Comment (pwittrock) [inline test/e2e_node/conformance_test.go]: Do you need to sleep 60 seconds?  This is going to extend the runtime of the tests by a good bit.

File: test/e2e_node/conformance_test.go [hunk 2]
```diff
@@ -99,66 +97,85 @@ var _ = Describe("Container Conformance Test", func() {
...
-				err := terminateCase.Create()
+			It("should be able to create, inspect and delete it [Conformance]", func() {
+				c := NewConformanceContainer(cl, *nodeName, api.Container{
+					Image:           "gcr.io/google_containers/busybox:1.24",
+					Name:            "terminate-container",
+					Command:         []string{"sh", "-c", "env"},
+					ImagePullPolicy: api.PullIfNotPresent,
+				})
+
+				By("create the container")
```
Comment (pwittrock) [inline test/e2e_node/conformance_test.go]: Thank you for all these messages.  They are very helpful.

File: test/e2e_node/conformance_test.go (hunk 1 above)
Comment (Random-Liu) [inline test/e2e_node/conformance_test.go]: Agree, I didn't remember where I copied this line from. Thanks! :0

File: pkg/kubelet/dockertools/manager.go [hunk 3]
```diff
@@ -384,11 +384,14 @@ func (dm *DockerManager) inspectContainer(id string, podName, podNamespace strin
 
 		terminationMessagePath := containerInfo.TerminationMessagePath
 		if terminationMessagePath != "" {
-			if path, found := iResult.Volumes[terminationMessagePath]; found {
-				if data, err := ioutil.ReadFile(path); err != nil {
-					message = fmt.Sprintf("Error on reading termination-log %s: %v", path, err)
-				} else {
-					message = string(data)
+			for _, mount := range iResult.Mounts {
```
Comment (pwittrock) [inline pkg/kubelet/dockertools/manager.go]: Why is this needed?

**Comment by pwittrock** [comment]
Review pass done.

File: pkg/kubelet/dockertools/manager.go (hunk 3 above)
Comment (Random-Liu) [inline pkg/kubelet/dockertools/manager.go]: This fixed the termination message path. (See #23658)
Only with it, the termination message path can work correctly, but unfortunately, it hasn't been merged.
To test termination message, we should let it work right first, so I cherry-pick that commit here. Will rebase after #23658 get merged.

Comment (pwittrock) [inline pkg/kubelet/dockertools/manager.go]: Does this mean termination message was broken and your test caught it?

Comment (pwittrock) [inline pkg/kubelet/dockertools/manager.go]: Does this mean termination message was broken and your test caught it?

Comment (Random-Liu) [inline pkg/kubelet/dockertools/manager.go]: I found it was broken before. The test proves that. :)
We don't have test for termination message before, in fact it was broken from 1.1, I just occasionally found it recently. With node e2e test, this tragedy would never happen again! :)

File: test/e2e_node/conformance_test.go (hunk 1 above)
Comment (Random-Liu) [inline test/e2e_node/conformance_test.go]: Will correct the time soon.

**Comment by Random-Liu** [comment]
Will rebase and self apply LGTM after #23658 gets merged.

**Comment by pwittrock** [comment]
SGTM

**Comment by Random-Liu** [comment]
The dependency #23658 has been merged. I've rebased the PR, self apply LGTM. :)

**Comment by Random-Liu** [comment]
Remove the LGTM for now to wait for @liangchenye's refactoring. :)

**Comment by liangchenye** [comment]
@Random-Liu 
The refactoring PR is here
https://github.com/kubernetes/kubernetes/pull/24329

**Comment by pwittrock** [comment]
@Random-Liu When you have a minute would you rebase?

**Comment by Random-Liu** [comment]
@pwittrock I was waiting for #24329, :) @liangchenye was refactoring the container conformance test, and I'll rebase all my PRs based on his refactoring PR.

**Comment by pwittrock** [comment]
Makes sense

**Comment by pwittrock** [comment]
@Random-Liu Let me know when I should TAL.  I assume you are busy for the next couple weeks with 1.3 stuff

**Comment by Random-Liu** [comment]
@pwittrock Will rebase these PRs after code freeze. :)

**Comment by Random-Liu** [comment]
@pwittrock Rebased the PR. Sorry for the delay. :)

**Comment by Random-Liu** [comment]
I just did a simple rebase, apply lgtm again.

**Comment by Random-Liu** [comment]
It's nice to have this test coverage in 1.3, so mark 1.3.

**Comment by goltermann** [comment]
E2E passed put check still listed as yellow and not in queue

@k8s-bot e2e test this issue: #IGNORE

**Comment by Random-Liu** [comment]
@k8s-bot test this issue #27663

**Comment by goltermann** [comment]
@k8s-bot node e2e test this

**Comment by goltermann** [comment]
@k8s-bot e2e test this issue: #IGNORE
